.tekton/pipeline/rhtap-staging-e2e.yaml

```diff
@@ -0,0 +1,264 @@
+          workspace: shared-file
+    - name: rebase-pr
+      taskRef:
+        name: rebase-pr
+      params:
+        - name: pull_request_number
+          value: $(params.pull_request_number)
+        - name: source_branch
+          value: $(params.source_branch)
+      workspaces:
+        - name: shared-file-path
+          workspace: shared-file
+      when:
+        - input: $(tasks.check-rebase-label.results.matched-rebase-label)
+          operator: in
+          values: ["True"]
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+      runAfter:
+        - check-rebase-label
+    - name: replace-revision
+      taskRef:
+        name: replace-revision
+      params:
+        - name: pr_branch_revision
+          value: $(params.revision)
+      workspaces:
+        - name: infra-deployment-git
+          workspace: git
+          subPath: infra-deployment-git
+        - name: shared-file-path
+          workspace: shared-file
+      when:
+        - input: $(tasks.check-rebase-label.results.matched-rebase-label)
+          operator: in
+          values: ["False"]
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+      runAfter:
+        - changed-component-app-name
+        - check-rebase-label
+    - name: clone-stage-e2e-git
+      taskRef:
+        name: git-clone
+      params:
+        - name: url
+          value: 'https://github.com/sawood14012/e2e-tests.git'
+        - name: deleteExisting
+          value: "true"
+        - name: revision
+          value: "verify-stage"
+      runAfter:
+        - check-merge-label
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+      workspaces:
+      - name: output
+        workspace: git
+        subPath: stage-e2e-git
+    - name: stage-verify-e2e-test
+      taskRef:
+        name: stage-verify-e2e-test
+      params:
+      - name: TASK_RETRIES
+        value: "$(context.pipelineTask.retries)"
+      workspaces:
+      - name: stage-e2e-git
+        workspace: git
+        subPath: stage-e2e-git
+      runAfter:
+        - replace-revision
+        - clone-stage-e2e-git
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+    - name: check-rebase-label-2
+      taskRef:
+        name: pr-rebase-labels
+      params:
+        - name: pull_request_number
+          value: $(params.pull_request_number)
+        - name: match_label
+          value: "needs-rebase"
+      runAfter:
+        - stage-verify-e2e-test
+      when:
+        - input: $(tasks.check-merge-label.results.matched-merge-label)
+          operator: in
+          values: ["True"]
+      workspaces:
+        - name: shared-file-path
+          workspace: shared-file
+    - name: remove-merge-label
```

Comment:
It might be useful to add a comment to some of these tasks to document the WHY.

E.g. `# Remove the 'ok-to-merge'  label on the PR so the final rebase does not trigger a new execution of the pipelinerun`